packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1107,6 +1090,86 @@ export class TransactionController extends BaseController<
     return this.nonceTracker.getNonceLock(address);
   }
 
+  private onBootCleanup() {
+    this.initApprovalsOfUnapprovedTransactions();
+    this.loadGasValuesOfUnapprovedTransactions();
+    this.submitApprovedTransactions();
+  }
+
+  /**
+   * Create approvals for all unapproved transactions on current chain.
+   */
+  private initApprovalsOfUnapprovedTransactions() {
```

Comment:
Minor, but is this more explicit as `createApprovalsForUnapprovedTransactions`?

Reply:
Fixed in https://github.com/MetaMask/core/pull/1916/commits/8b11fe7ca9352e56a3f261ba3c88fff69b0f06a9